src/duckxiterator.hpp

```diff
@@ -0,0 +1,43 @@
+/*
+ * Under MIT license
+ * Author: Amir Mohamadi (@amiremohamadi)
+ * DuckX is a free library to work wirh docx files.
+ */
+
+#ifndef DUCKXITERATOR_H
+#define DUCKXITERATOR_H
+
+namespace duckx {
+	template <class T>
+	class Iterator {
+	private:
+		T* _obj = nullptr;
+
+	public:
+		Iterator() = default;
+		Iterator(T& obj) : _obj(&obj) {}
+
+		// We only care comparator against end.
+		bool operator!=(const Iterator&) const { return _obj != nullptr; }
+
+		// Check against the end
+		void operator++() {
```

Comment:
Can we satisfy all requirements for https://en.cppreference.com/w/cpp/named_req/InputIterator

Reply:
I will check it over the weekend.

Reply:
I think we will need a way to check the contents of the nodes against one another first, in order to actually compare two values. Currently we can only check against the memory location, which doesn't check the actual content.

Reply:
@amiremohamadi 
Could we get a comparison operator to check the contents so that the iterators can be compared to each other? (just check against current and the next node? or more to do?)
Could we change has_next() functions to be const? (seems trivial)

If nobody is up to the task, I can make a separate PR for these over the weekend and submit.

Reply:
@CihanSari My idea is to make **DuckX** iterators like iterators in [**pugixml**](https://github.com/zeux/pugixml).
Since the structure of DuckX files may change soon, please put iterator classes in to "duckx.hpp" and "duckx.cpp".

Feel free to create a new PR! :smiley: